packages/frontend/core/src/components/attachment-viewer/worker/worker.ts

```diff
@@ -0,0 +1,171 @@
+import { DebugLogger } from '@affine/debug';
+import type { Document } from '@toeverything/pdf-viewer';
+import {
+  createPDFium,
+  PageRenderingflags,
+  Runtime,
+  Viewer,
+} from '@toeverything/pdf-viewer';
+
+import type { MessageData, MessageDataType } from './types';
+import { MessageOp, MessageState, RenderKind } from './types';
+import { resizeImageBitmap } from './utils';
+
+const logger = new DebugLogger('affine:worker:pdf');
+
+let inited = false;
+let viewer: Viewer | null = null;
+let doc: Document | undefined = undefined;
+
+const canvas = new OffscreenCanvas(0, 0);
+const ctx = canvas.getContext('2d');
+const cached = new Map<number, ImageBitmap>();
```

Comment:
may need to have a gc for the cache

Reply:
WeakMap?

Reply:
I just realized the worker is used per Viewer component instance.

Reply:
Yes, each is a separate instance